File: Chapter1/R/Quartiles.R
game_points <- c(35,56,43,59,63,79,35,41,64,43,93,60,77,24,82)

# Varianza
dt_var = var(game_points)
print(round(dt_var,2))

# Desviación Standard
dt_std = sd(game_points); print(round(dt_std,2))

#Rango
range_val<-function(x) return(diff(range(x)))
dt_range = range_val(game_points)
print(dt_range)

dt_quantile = quantile(game_points,probs = c(0.2,0.8,1.0))
print(dt_quantile)
dt_iqr = IQR(game_points); print(dt_iqr)

File: Chapter1/Python/Quartiles.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon Jun  7 01:11:16 2021

@author: moisesr
"""
import numpy as np
from scipy import stats
from statistics import variance, stdev


game_points = np.array([35,56,43,59,63,79,35,41,64,43,93,60,77,24,82])

# Calculare Variance
#This is the mean of squared deviations from the mean
dt_var = variance(game_points)
print("Sample Variance :" ,round(dt_var,2))

# Calculate Stadard Deviation
#This is the square root of variance
dt_std = stdev(game_points)
print("Sample std.dev :", round(dt_std,2))

# Calculate Range
dt_rng = np.max(game_points, axis=0) - np.min(game_points, axis=0)
print("Range :", dt_rng)

# Calculate the percentiles
print("Quantiles :")
for val in [20,80,100]:
    dt_qnls = np.percentile(game_points,val)
    print(str(val)+"%", dt_qnls)

# Calculate IQR
q75, q25 = np.percentile(game_points, [75,25])
print("Interquantile range :",q75 - q25)